### PR TITLE
Adding support for yolov8-seg

### PR DESCRIPTION
**Pull request by edenlum**
## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [ ] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

File: tutorials/quick_start/pytorch_fw/ultralytics_lib/replacers.py [hunk 1]
```diff
@@ -133,6 +134,49 @@ def replace(self, model):
         return replace_2d_deg_module(model, Detect, self.get_new_module, self.get_config)
 
 
+class SegmentReplacer(Segment):
+    """
+    Replaces the Segment module to use the replaced Detect forward function.
```
Comment (Idan-BenAmi) [inline tutorials/quick_start/pytorch_fw/ultralytics_lib/replacers.py:139]: mention also about the output split, in order to improve the quantization.

File: tutorials/quick_start/pytorch_fw/ultralytics_lib/replacers.py [hunk 2]
```diff
@@ -133,6 +134,49 @@ def replace(self, model):
...
+        nm = c.nm
+        npr = c.npr
+        ch = [next(next(x.children()).children()).in_channels for x in c.cv2.children()]
+        return [nc, nm, npr, ch]
+
+    def replace(self, model):
+        return replace_2d_deg_module(model, Segment, self.get_new_module, self.get_config)
+
+
+ForwardReplacer = {
```
Comment (Idan-BenAmi) [inline tutorials/quick_start/pytorch_fw/ultralytics_lib/replacers.py:175]: I would locate this dictionary at the top of the file (before the class definitions) and rename it to "TaskModuleReplacer".

File: tutorials/quick_start/pytorch_fw/ultralytics_lib/replacers.py [hunk 3]
```diff
@@ -192,6 +236,42 @@ def postprocess(self, preds):
...
+        y = torch.cat((dbox, y_cls), 1)
+        # additional part for segmentation
+        preds = (torch.cat([y, masks_coeffs], 1), (y_cls, masks_coeffs, proto))
+
+        # Original post-processing part
+        preds = super().postprocess(preds)
+
+        return preds
+
+TASK_MAP = {
```
Comment (Idan-BenAmi) [inline tutorials/quick_start/pytorch_fw/ultralytics_lib/replacers.py:265]: Check if you can import it from ultralytics instead of re-define it. If not, I would minimize it and keep only the relevant fields. Also, locate at the top of the file.

**Review by Idan-BenAmi — CHANGES_REQUESTED** [review]

**Review by Idan-BenAmi — APPROVED** [review]
